buildcontext/gitlookup.go

```diff
@@ -306,22 +306,26 @@ func (gl *GitLookup) getHostKeyAlgorithms(hostname string) ([]string, []string,
 	if err != nil {
 		gl.console.Warnf("failed to load ~/.ssh/known_hosts: %s", err)
 	}
+	gl.console.VerbosePrintf("loaded %d key(s) from known_hosts and %d default key(s)", len(knownHostsKeyScans), len(defaultKeyScans))
+
+	foundKeys := make(map[string]bool)
 	for _, keyScans := range [][]string{
 		knownHostsKeyScans,
 		defaultKeyScans,
 	} {
 		for _, keyScan := range keyScans {
 			keyAlg, keyData, err := parseKeyScanIfHostMatches(keyScan, hostname)
-			switch err {
-			case nil:
-			case errKeyScanNoMatch:
```

Comment:
this is no longer being explicitly checked, when a `errKeyScanNoMatch` "error" occurs, we should still do the `continue`, but printing out a warning saying the key failed to parse is incorrect (since it did parse, it's just being omitted due to it not matching).

When it is skipped, can we do

```
gl.console.VerbosePrintf("ignoring key scan %q: due to host mismatch", keyScan)
```